File: Quick_sort.py
def quicksort(array, inicio, fin):
    if inicio >= fin:
        return  # Condición de parada

    # Elección del pivote
    pivote = array[inicio + (fin - inicio) // 2]  # Valor del pivote
    i = inicio
    j = fin

    while i <= j:
        # Encontrar un elemento a la izquierda que sea mayor o igual al pivote
        while array[i] < pivote:
            i += 1
        # Encontrar un elemento a la derecha que sea menor o igual al pivote
        while array[j] > pivote:
            j -= 1
        # Intercambiar elementos si i <= j
        if i <= j:
            array[i], array[j] = array[j], array[i]  # Intercambio usando una tupla
            i += 1
            j -= 1

    # Recursivamente ordenar las dos sub-particiones
    if inicio < j:
        quicksort(array, inicio, j)  # Partición izquierda
    if i < fin:
        quicksort(array, i, fin)  # Partición derecha


def lectura(filename):
    numeros = []
    try:
        # Abre el archivo en modo lectura
        with open(filename, "r") as archivo:
            # Lee el archivo línea por línea y convierte cada línea a entero
            for linea in archivo:
                numeros.append(int(linea.strip()))  # Convierte y agrega a la lista
    except FileNotFoundError:
        print("No se pudo abrir el archivo.")
    return numeros


def main():
    filename = "C:\\Users\\Usuar\\Desktop\\practica ADA\\random\\los100000.txt"
    array = lectura(filename)

    # Ordenar el array
    if array:
        quicksort(array, 0, len(array) - 1)

        # Imprimir el array ordenado
        print(" ".join(map(str, array)))
    else:
        print("El archivo está vacío o no se leyeron números.")


if __name__ == "__main__":
    main()

File: Quick_sort.java
import java.io.*;
import java.util.ArrayList;
import java.util.List;

public class Quick_sort {

    // Función para implementar QuickSort
    public static void quicksort(List<Integer> array, int inicio, int fin) {
        if (inicio >= fin) return; // Condición de parada
        
        // Elección del pivote
        int pivote = array.get(inicio + (fin - inicio) / 2); // Valor del pivote (elemento en la mitad)
        int i = inicio;
        int j = fin;
        
        while (i <= j) {
            // Encontrar un elemento a la izquierda que sea mayor o igual al pivote
            while (array.get(i) < pivote) {
                i++;
            }
            // Encontrar un elemento a la derecha que sea menor o igual al pivote
            while (array.get(j) > pivote) {
                j--;
            }
            // Intercambiar elementos si i <= j
            if (i <= j) {
                int temp = array.get(i);
                array.set(i, array.get(j));
                array.set(j, temp);
                i++;
                j--;
            }
        }

        // Recursivamente ordenar las dos sub-particiones
        if (inicio < j) {
            quicksort(array, inicio, j); // Partición izquierda
        }
        if (i < fin) {
            quicksort(array, i, fin); // Partición derecha
        }
    }

    // Función para leer el archivo y llenar la lista
    public static void lectura(List<Integer> numeros) {
        BufferedReader reader;
        String linea;

        try {
            // Abre el archivo en modo lectura
            reader = new BufferedReader(new FileReader("C:\\Users\\Usuar\\Desktop\\practica ADA\\random\\los100000.txt"));

            // Verifica si el archivo se abrió correctamente y lee línea por línea
            while ((linea = reader.readLine()) != null) {
                // Convierte la línea leída a entero y la almacena en la lista
                numeros.add(Integer.parseInt(linea));
            }
            reader.close();  // Cierra el archivo

        } catch (IOException e) {
            System.out.println("No se pudo abrir el archivo.");
        }
    }

    // Función principal
    public static void main(String[] args) {
        List<Integer> array = new ArrayList<>();
        lectura(array);

        // Verificar si se leyeron números
        if (!array.isEmpty()) {
            // Llamar a quicksort
            quicksort(array, 0, array.size() - 1);

            // Imprimir el array ordenado
            for (int num : array) {
                System.out.print(num + " ");
            }
            System.out.println();
        } else {
            System.out.println("El archivo está vacío o no se leyeron números.");
        }
    }


}
